rq/queue.py

```diff
@@ -329,42 +410,46 @@ def enqueue_call(self, func, args=None, kwargs=None, timeout=None,
             retry=retry
         )
 
-        # If a _dependent_ job depends on any unfinished job, register all the
-        # _dependent_ job's dependencies instead of enqueueing it.
-        #
-        # `Job#fetch_dependencies` sets WATCH on all dependencies. If
-        # WatchError is raised in the when the pipeline is executed, that means
-        # something else has modified either the set of dependencies or the
-        # status of one of them. In this case, we simply retry.
-        if depends_on is not None:
-            with self.connection.pipeline() as pipe:
-                while True:
-                    try:
-
-                        pipe.watch(job.dependencies_key)
-
-                        dependencies = job.fetch_dependencies(
-                            watch=True,
-                            pipeline=pipe
-                        )
-
-                        pipe.multi()
-
-                        for dependency in dependencies:
-                            if dependency.get_status(refresh=False) != JobStatus.FINISHED:
-                                job.set_status(JobStatus.DEFERRED, pipeline=pipe)
-                                job.register_dependency(pipeline=pipe)
-                                job.save(pipeline=pipe)
-                                job.cleanup(ttl=job.ttl, pipeline=pipe)
-                                pipe.execute()
-                                return job
-
-                        break
-                    except WatchError:
-                        continue
+        return self.setup_dependencies_and_enqueue(
+            job,
+            pipeline=pipeline,
+            at_front=at_front
+        )
 
-        job = self.enqueue_job(job, at_front=at_front)
-        return job
+    def enqueue_many(
+        self,
+        job_datas,
+        pipeline=None
```

Comment:
Do we need to accept `pipeline` argument here?

Reply:
@selwin I did for completeness sake, especially with the original motivation of this PR was adding the ability to pipeline the various enqueue methods, so it was easy enough to implement this.